system/jlib/jstats.h

```diff
@@ -114,15 +115,20 @@ interface IStatisticCollection : public IInterface
     virtual unsigned getNumStatistics() const = 0;
     virtual bool getStatistic(StatisticKind kind, unsigned __int64 & value) const = 0;
     virtual void getStatistic(StatisticKind & kind, unsigned __int64 & value, unsigned idx) const = 0;
+    virtual bool getStatisticSum(StatisticKind kind, unsigned __int64 & value) const = 0;
     virtual IStatisticCollectionIterator & getScopes(const char * filter, bool sorted) = 0;
     virtual void getMinMaxScope(IStringVal & minValue, IStringVal & maxValue, StatisticScopeType searchScopeType) const = 0;
     virtual void getMinMaxActivity(unsigned & minValue, unsigned & maxValue) const = 0;
+    virtual bool setStatistic(const char *scope, StatisticKind kind, unsigned __int64 value) = 0;
     virtual void serialize(MemoryBuffer & out) const = 0;
     virtual unsigned __int64 queryWhenCreated() const = 0;
     virtual void mergeInto(IStatisticGatherer & target) const = 0;
     virtual StringBuffer &toXML(StringBuffer &out) const = 0;
     virtual void visit(IStatisticVisitor & target) const = 0;
     virtual void visitChildren(IStatisticVisitor & target) const = 0;
+    virtual void refreshAggregates(const StatisticsMapping & mapping, std::function<void(const char * scope, StatisticScopeType sst, StatisticKind kind, stat_type value)> & fWhenAggregateUpdated) = 0;
```

Comment:
This callback definition (std::function<void(const char * scope, StatisticScopeType sst, StatisticKind kind, stat_type value)>) would benefit from a typedef to clarify.